colors/nord.vim

```diff
@@ -646,6 +646,37 @@ hi! link jsNoise Delimiter
 hi! link jsPrototype Keyword
 hi! link jsRegexpString SpecialChar
 
+" TypeScript
+" > HerringtonDarkholme/yats.vim
+call s:hi("typescriptDecorator", s:nord12_gui, "", s:nord12_term, "", "NONE", "")
+hi! link typescriptOperator Operator
+hi! link typescriptBinaryOp Operator
+hi! link typescriptAssign Operator
+hi! link typescriptMember Identifier
+hi! link typescriptDOMStorageMethod Identifier
+hi! link typescriptArrowFuncArg Identifier
+hi! link typescriptGlobal Function
+hi! link typescriptBOMWindowProp Function
+hi! link typescriptClassName Function
+hi! link typescriptArrowFuncDef Function
+hi! link typescriptAliasDeclaration Function
+hi! link typescriptPredefinedType Function
+hi! link typescriptTypeReference Function
+hi! link typescriptTypeAnnotation Function
+hi! link typescriptDocNamedParamType SpecialComment
+hi! link typescriptDocNotation Keyword
+hi! link typescriptDocTags Keyword
+hi! link typescriptImport Keyword
+hi! link typescriptExport Keyword
+hi! link typescriptTry Keyword
+hi! link typescriptVariable Keyword
+hi! link typescriptBraces Normal
+hi! link typescriptObjectLabel Normal
+hi! link typescriptCall Normal
```

Comment:
Added seven more highlighting groups:

1. `typescriptClassHeritage` ensures the name of classes and interfaces after `implements` and `extends` are styled properly (`nord7` for structs/classes).
   <p align="center"><img src="https://user-images.githubusercontent.com/7836623/84700900-4bad8c80-af54-11ea-9caa-2413f53af164.png" width="80%" /></p>
2. `typescriptFuncTypeArrow` corrects the color of _fat arrows_ to use `nord9` instead of `nord8` since they are kind of operators/keywords.
   <p align="center"><img src="https://user-images.githubusercontent.com/7836623/84700311-3d12a580-af53-11ea-8102-39c86eb54df9.png" width="80%" /></p>
3. `typescriptMemberOptionality` the targets the `?` question mark character that marks properties are optional in interface/type definitions.
   <p align="center"><img src="https://user-images.githubusercontent.com/7836623/84699121-43a01d80-af51-11ea-8ca2-7560791e04f5.png" width="40%" /></p>
4. `typescriptNodeGlobal` are global _Node.js_ elements like `process` that can also be seen as a kind of _static_ struct/class. 
5. `typescriptTypeBrackets` ensures `<>` brackets of types are highlighted as structural elements (`nord9`).
   <p align="center"><img src="https://user-images.githubusercontent.com/7836623/84701261-f2922880-af54-11ea-9779-ba1314480921.png" width="80%" /></p>
6. `tsxIntrinsicTagName` are “normal“ HTML elements tags like `<div>` that should match Nord's style for HTML.
7. Even when React elements are kind of custom HTML elements, the `tsxTagName` highlight group should still use the same color like for “normal“ HTML tags since they can be distinguished due to their large and lower case naming.
   <p align="center"><img src="https://user-images.githubusercontent.com/7836623/84701503-66cccc00-af55-11ea-80b7-9793fc894284.png" width="60%" /></p>

```suggestion
hi! link typescriptCall Normal
hi! link typescriptClassHeritage typescriptClassName
hi! link typescriptFuncTypeArrow Structure
hi! link typescriptMemberOptionality Structure
hi! link typescriptNodeGlobal typescriptGlobal
hi! link typescriptTypeBrackets Structure
hi! link tsxIntrinsicTagName htmlTag
hi! link tsxTagName tsxIntrinsicTagName
```

Reply:
I can't remember that I faced the `tsxTagName` syntax element when examined .tsx files. Every tag was either `jsxTagNme` or `jsxComponentName`.

Reply:
I faced the exact opposite with only `tsx*` groups 😮

Have you made sure to use a clean/minimal `vimrc´ without any other active being installed/loaded plugins?
Can you also please test whats the output when placing the cursor into a JSX tag within a `.tsx` file and then running `echo map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")')`?

Reply:
First I removed everything from my vimrc except `sheerun/vim-polyglot` and `arcticicestudio/nord-vim`. Get the same names for components `jsxTagNme` and `jsxComponentName`.

After I removed `sheerun/vim-polyglot`  I began to get "normal" `tsx*` names.

Looks like something in the polyglot plugin renames elements syntax name (not sure how to call them properly).

Reply:
I'm going to accept your changes. I can send a fix for my case with the polyglot plugin in a separate PR.

Reply:
That's a problem users often come across since such “all-in-one“ packages are known to cause problems and often add much overhead that'll never be used anyway. It's hard to debug problems since it is almost impossible to trace back which of the many plugins broke another one. I can only recommend to add plugins that are really needed, e.g. for programming languages used on a daily basis.
 
I guess the highlighting for the _JSX_ elements is caused by the [MaxMEllon/vim-jsx-pretty][r] plugin that is included in `sheerun/vim-polyglot`. This plugin also additionally defines highlighting groups for TypeScript and _TSX_ so it's kind of unpredictable which plugin takes precedence when both are loaded, either `HerringtonDarkholme/yats.vim` or `MaxMEllon/vim-jsx-pretty`.

[r]: https://github.com/MaxMEllon/vim-jsx-pretty/blob/master/after/syntax/jsx_pretty.vim#L186-L189

Reply:
I see. Thanks for taking time and explaining this to me! I'll follow your advice and replace `vim-polyglot` plugin.